server/index.js

```diff
@@ -63,6 +62,105 @@ app.get('/api/products/:productId', (req, res, next) => {
     });
 });
 
+app.get('/api/cart/', (req, res) => {
+  const sql = `
+    SELECT "c"."cartItemId",
+            "c"."price",
+            "p"."productId",
+            "p"."image",
+            "p"."name",
+            "p"."shortDescription"
+        FROM "cartItems" AS "c"
+        JOIN "products" AS "p" USING ("productId")
+      WHERE "c"."cartId" = $1
+  `;
+  const value = [req.session.cartId];
+  db.query(sql, value)
+    .then(result => {
+      res.json(result.rows);
+    })
+    .catch(err => {
+      console.error(err);
+      return res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.post('/api/cart/', (req, res, next) => {
+  const { productId, price } = req.body;
+  if (!Number.isInteger(productId) || productId <= 0) {
+    next(new ClientError('"productId" must be a positive integer', 400));
+  }
+  const sqlInitialQuery = `
+    SELECT "productId"
+    FROM "products"
+    WHERE "productId" = $1
+  `;
+  const valueInitial = [productId];
+  db.query(sqlInitialQuery, valueInitial)
+    .then(result => {
+      if (!result.rows.length) {
+        res.status(400).json({
+          error: `Cannot find the productId:" ${productId}`
+        });
+      }
+      if (!req.session.cartId) {
+        const sqlForCreateCart = `
+          INSERT INTO "carts" ("cartId", "createdAt")
+          VALUES (default, default)
+          RETURNING "cartId"
+        `;
+        return db.query(sqlForCreateCart)
+          .then(result => {
+            return { cartId: result.rows[0].cartId, price: price };
+          });
+      } else {
+        return { cartId: req.session.cartId, price: price };
+      }
+    })
+    .then(result => {
+      req.session.cartId = result.cartId;
+      req.session.save();
+      const sqlForCreateCartItem = `
+          INSERT INTO "cartItems" ("cartId", "productId", "price")
+          VALUES ($1, $2, $3)
+          RETURNING "cartItemId"
+        `;
+      const valuesForCartItem = [req.session.cartId, productId, Number(price)];
+      return db.query(sqlForCreateCartItem, valuesForCartItem)
+        .then(result => {
+          return { cartItemId: result.rows[0].cartItemId };
+        });
+    })
+    .then(result => {
+      req.session.cartItemId = result.cartItemId;
+      req.session.save();
+      const sqlForRetriveData = `
+            SELECT "c"."cartItemId",
+                    "c"."price",
+                    "p"."productId",
+                    "p"."image",
+                    "p"."name",
+                    "p"."shortDescription"
+               FROM "cartItems" AS "c"
+               JOIN "products" AS "p" USING ("productId")
+              WHERE "c"."cartItemId" = $1
+            `;
+      const valuesForRetriveData = [req.session.cartItemId];
+      db.query(sqlForRetriveData, valuesForRetriveData)
```

Comment:
Be sure to `return` this `db.query()` so that if it fails, the `.catch()` can handle the error.

Reply:
Added `return` statement